### PR TITLE
[jwplayer] Add type definitions for configuration options

### PR DESCRIPTION
**Pull request by djcsdy**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://developer.jwplayer.com/jw-player/docs/developer-guide/customization/configuration-reference/
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/jwplayer/index.d.ts [hunk 1]
```diff
@@ -317,6 +317,329 @@ declare namespace jwplayer {
     type NoParamEvent = 'adBlock' | 'beforeComplete' | 'complete' | 'beforePlay' | 'displayClick' | 'playlistComplete'
             | 'seeked' | 'remove';
 
+    type RequiredConfiguration = {
+        readonly file: string;
+    } | {
+        readonly playlist: PlaylistConfiguration;
+    };
+
+    type Configuration = RequiredConfiguration & {
```
Comment (SimonSchick) [inline types/jwplayer/index.d.ts]: Shouldn't this be an interface?

File: types/jwplayer/index.d.ts [hunk 2]
```diff
@@ -317,6 +317,329 @@ declare namespace jwplayer {
     type NoParamEvent = 'adBlock' | 'beforeComplete' | 'complete' | 'beforePlay' | 'displayClick' | 'playlistComplete'
             | 'seeked' | 'remove';
 
+    type RequiredConfiguration = {
+        readonly file: string;
```
Comment (SimonSchick) [inline types/jwplayer/index.d.ts]: I think your use of `readonly` is excessive and will make the actual typings hard to use, please remove those.

**Review by SimonSchick — CHANGES_REQUESTED** [review]
I strongly suggest breaking interface apart further, making the `Configuration` one huge monolithic type seems like a bad idea.

File: types/jwplayer/index.d.ts (hunk 2 above)
Comment (djcsdy) [reply types/jwplayer/index.d.ts]: None of these values are written to by JWPlayer, so they should be `readonly`.

File: types/jwplayer/index.d.ts (hunk 1 above)
Comment (djcsdy) [reply types/jwplayer/index.d.ts]: The purpose of this arrangement is to require either the property `playlist` or `file` to be set, but not both.

**Comment by djcsdy** [comment]
I agree that it would be a good idea to break `Configuration` up into smaller parts. I wanted to do so myself. However I don’t have unlimited time, and this is unambiguously better than `any` so I’m submitting it. If you want to shit on my decisions feel free to make changes yourself or pay me.

**Comment by SimonSchick** [comment]
@djcsdy I am not *shit*ting on your decision, but since you feel offended by my constructive criticism I will leave it up to someone else to review these changes.

**Comment by djcsdy** [comment]
There is a big difference between constructive criticism and ordering people about. I found nothing helpful in your remarks.

**Comment by djcsdy** [comment]
I can’t be bothered with this any more.

**Pull request closed without merging**